src/util/file_log.rs

```diff
@@ -91,21 +91,24 @@ impl RotatingFileLogger {
         self.next_rotation_time = compute_rotation_time(&now, self.rotation_timespan);
     }
 
+    /// Flushes and closes log file, without rotation.
     fn close(&mut self) -> io::Result<()> {
-        self.file.flush()
+        self.file.as_mut().unwrap().flush()?;
```

Comment:
You can just `self.file.take().unwrap().flush()?`.
BTW I think here `if let Some(..)` is better, because `close` is a public method. It should be safe for closing it multiple times.